catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -69,45 +75,59 @@ public CatalogBackupPlugin() {
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
     public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+
         if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Performing backup of metacards in CreateResponse.");
+            LOGGER.trace("Performing backup of metacards in CreateResponse.");
 
             if (rootBackupDir == null) {
                 throw new PluginExecutionException("No root backup directory configured.");
             }
 
-            List<String> errors = new ArrayList<String>();
+            EXECUTOR_SERVICE.submit(() -> processBatch(input.getCreatedMetacards()));
```

Comment:
Correct, you need to trigger off of a particular batch size and a period of time. So for example, if you've queued up 2000 metacards, _start doing the backup_ *OR* if has been 10 minutes since the last backup and you have metacards to back up, _start doing the backup_.